### DISCUSSION TITLE
How to use `astra_mex_algorithm("get_res_norm")`

### DISCUSSION THREAD

**Discussion by eigenvivek** [astra-toolbox/astra-toolbox#486] [Q&A] [answered]
For 3D reconstruction, [the documentation](https://astra-toolbox.com/docs/algs/CGLS3D_CUDA.html#extra-features) says that `SIRT3D_CUDA` and `CGLS3D_CUDA` support `astra_mex_algorithm("get_res_norm")`. 

Is this an additional loss term that can be used during the optimization?  If yes, how do you enable this option with the Python API?

I've tried searching the GitHub repository for code that calls this option, but wasn't able to find an example. Thank you for your help!

**Comment by askorikov** [comment, marked as answer]
Hi! This is actually the primary loss that SIRT and CGLS algorithms minimize. With this option, you can get its value at each iteration. See https://github.com/astra-toolbox/astra-toolbox/blob/f314cc70dad3f2ebbc9a0f51ce41ac15218f09fd/samples/python/s016_plots.py#L68

**Reply by eigenvivek** [reply]
Ah got it, thank you!
